File: javascript/test/instance/EmptyCollectionHolderForTest.ts
//··························································
// Copyright (c) 2023-2025. Jonathan Bédard ~ JóôòKiwi
//
// This project is free to use.
// All the right is reserved to the author of this project.
// My projects:
//  - https://github.com/joooKiwi/type
//  - https://github.com/joooKiwi/lazy
//  - https://github.com/joooKiwi/collection
//  - https://github.com/joooKiwi/enumeration
//··························································

import type {StraightCollectionHolderForTest} from "./StraightCollectionHolderForTest"

import {EmptyCollectionHolder} from "../../src/EmptyCollectionHolder"

export class EmptyCollectionHolderForTest
    extends EmptyCollectionHolder
    implements StraightCollectionHolderForTest<never> {

    public constructor() { super() }

    public get amountOfCall(): never {
        throw new Error("The amountOfCall will never change on an EmptyCollectionHolder.",)
    }

    public execute(action?: (instance: this,) => void,): never
    public execute() {
        throw new Error("An EmptyCollectionHolder does not need to execute anything since all of the methods are self contained.",)
    }

}

File: java/collection.joookiwi/src/test/java/instance/EmptyCollectionHolderForTest.java
package instance;

import java.util.function.Consumer;
import joookiwi.collection.java.EmptyCollectionHolder;
import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNullByDefault;
import org.jetbrains.annotations.Nullable;

import static joookiwi.collection.java.CommonContracts.ALWAYS_FAIL_0;
import static joookiwi.collection.java.CommonContracts.ALWAYS_FAIL_1;

@NotNullByDefault
public final class EmptyCollectionHolderForTest<T extends @Nullable Object>
        extends EmptyCollectionHolder<T>
        implements StraightCollectionHolderForTest<T, EmptyCollectionHolderForTest<T>> {

    public EmptyCollectionHolderForTest() { super(); }

    @Contract(ALWAYS_FAIL_0)
    @Override public int getAmountOfCall() { throw new RuntimeException("The amountOfCall will never change on an EmptyCollectionHolder"); }

    @Contract(ALWAYS_FAIL_1)
    @Override public EmptyCollectionHolderForTest<T> execute(final @Nullable Consumer<EmptyCollectionHolderForTest<T>> action) { throw new RuntimeException("An EmptyCollectionHolder does not need to execute anything since all of the methods are self contained."); }

}
